lib/google_search_data_viewer/keyword/search_results.ex

```diff
@@ -7,4 +7,20 @@ defmodule GoogleSearchDataViewer.Keyword.SearchResults do
     |> SearchResultUrl.changeset()
     |> Repo.insert()
   end
+
+  def group_by_adword_types(search_result_urls) do
+    search_result_urls
+    |> Enum.group_by(&{&1.is_adword, &1.is_top_adword})
+    |> Enum.map(&add_adword_type_from_group/1)
+    |> Map.new()
+  end
+
+  defp add_adword_type_from_group({{true, true}, search_result_urls}),
+    do: {:top_adwords, search_result_urls}
+
+  defp add_adword_type_from_group({{false, false}, search_result_urls}),
+    do: {:non_adwords, search_result_urls}
+
+  defp add_adword_type_from_group({{true, false}, search_result_urls}),
+    do: {:adwords, search_result_urls}
```

Comment:
I originally had the following function which **only** calclates the counts / summarized data for the URLs:

```def get_url_stats_for_search_results(search_results) do
    search_results
    |> Enum.group_by(&{&1.is_adword, &1.is_top_adword})
    |> Enum.map(fn {k, v} ->
      add_adword_count_to_url_stats(%{
        is_adword: elem(k, 0),
        is_top_adword: elem(k, 1),
        count: Enum.count(v)
      })
    end)
    |> Map.new()
  end


defp add_adword_count_to_url_stats(%{is_adword: true, is_top_adword: true, count: count}) do
    {:total_top_adwords, count}
  end

  defp add_adword_count_to_url_stats(%{is_adword: false, is_top_adword: false, count: count}) do
    {:total_non_adwords, count}
  end

  defp add_adword_count_to_url_stats(%{is_adword: true, is_top_adword: false, count: count}) do
    {:total_adwords, count}
  end
  ```
  
  But as mentioned in the PR description and code, I chose to return just the list of URLs `grouped` by the type rather than the List of URLs and the counts because the counts can be calculated on the frontend as they are grouped already.
  
I did this because it only requires one DB hit instead of two, and I don't see a current or future requirement to obtain just the summary/count data on its own. Let me know what you think about this too, thanks.